configs/phpmd.xml

```diff
@@ -2,69 +2,45 @@
 <ruleset name="Drupal ruleset"
          xmlns="http://pmd.sourceforge.net/ruleset/2.0.0"
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-         xsi:schemaLocation="http://pmd.sourceforge.net/ruleset/2.0.0 http://pmd.sourceforge.net/ruleset_2_0_0.xsd">
+         xsi:schemaLocation="http://pmd.sourceforge.net/ruleset/2.0.0
+                http://pmd.sourceforge.net/ruleset_2_0_0.xsd"
+         xsi:noNamespaceSchemaLocation="
+                http://pmd.sf.net/ruleset_xml_schema.xsd">
 
-  <description>PMD ruleset for Drupal.</description>
+  <description>
+    Ruleset for PHPMD analysis of Drupal projects. Excludes coding issues
+    handled better by PHPCS and rules which have too many false positives
+    in a typical Drupal codebase.
+  </description>
 
   <rule ref="rulesets/cleancode.xml">
-    <exclude name="BooleanArgumentFlag" />
-    <exclude name="ElseExpression" />
-    <exclude name="IfStatementAssignment" />
+    <!--
+      Validated by PHPCS and is in conflict with the DrupalPractices sniffs.
+    -->
     <exclude name="MissingImport" />
+    <!--
+      PHPMD is not smart enough to see the difference between static access and
+      named constructors.
+    -->
     <exclude name="StaticAccess" />
   </rule>
 
-  <rule ref="rulesets/codesize.xml">
-    <exclude name="CyclomaticComplexity" />
-    <exclude name="ExcessiveClassComplexity" />
-    <exclude name="ExcessiveMethodLength" />
-    <exclude name="ExcessiveParameterList" />
-    <exclude name="NPathComplexity" />
-    <exclude name="TooManyMethods" />
-    <exclude name="TooManyPublicMethods" />
-  </rule>
-
-  <rule ref="rulesets/codesize.xml/CyclomaticComplexity">
-    <properties>
-      <property name="reportLevel" value="20" />
-    </properties>
-  </rule>
-
-  <rule ref="rulesets/codesize.xml/ExcessiveMethodLength">
+  <rule ref="rulesets/codesize.xml" />
+  <!--rule ref="rulesets/codesize.xml/ExcessiveMethodLength">
     <properties>
       <property name="minimum" value="250" />
     </properties>
-  </rule>
-
-  <rule ref="rulesets/codesize.xml/TooManyPublicMethods">
```

Comment:
Breek je hiermee niet de helft van de value objects?

Reply:
Never had issues with them... If a value object has to many getters it probably should group some of them together in another Value object.

eg:

```php
$person->getFirstName;
$person->getLastName;
$person->getStreet();
$person->getStreetNumber();
```

Should become:
```php
$person->getName()->getFirstName();
$person->getName()->getLastName();
$person->getAddress()->getStreet();
$person->getAddress()->getHouseNumber();
$person->getAddress()->getLocality()->getZipCode():
$person->getAddress()->getLocality()->getName();
```

Reply:
Okay, but this conflicts with the form builder as well If your form has a could of ajax handlers (results in +2 methods each) you'll hit this error as well.